pkg/logger/handler.go

```diff
@@ -105,15 +136,18 @@ func (eh *LoggerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// Proxy Request
 	r.Body = io.NopCloser(bytes.NewBuffer(body))
-	rr := httptest.NewRecorder()
-	eh.next.ServeHTTP(rr, r)
-	responseBody := rr.Body.Bytes()
-	contentType = rr.Header().Get("Content-Type")
-	if contentType != "" {
-		w.Header().Set("Content-Type", contentType)
+	// TODO: Set a reasonable intial buffer size
+	var responseBuf bytes.Buffer
+	lrw := &loggingResponseWriter{ResponseWriter: w, responseBuffer: &responseBuf, log: eh.log}
+	eh.next.ServeHTTP(lrw, r)
+	// Read the response body from the buffer
+	reader := bufio.NewReader(lrw.responseBuffer)
+	responseBody, err := io.ReadAll(reader)
```

Comment:
Currently, We are collecting all the streamed data and sending it as a single cloudevent to the logger (that's how it was done previously so I am just keeping it as it is). Should we send each chunck as a separate cloudevent ? This will also reduce the memory footprint of agent sidecar.

Reply:
So previously we had the full eventstream response body, in the cloudevent?
I didn't realize that.  Useful.

A single cloudevent is certainly easier to process from a consumer perspective, but multiple events (as long as it is possible to interpret it as multiple response events for the same stream and put it back together when analyzing the cloudevents).

Reply:
We have [cloudevent id](https://github.com/kserve/kserve/pull/3847/files#diff-d6234b161cbbad078660d85ab2911474143501480664e5e5ca2395d25c4378b3R157) which allows us to identify that a specific stream is belongs to a certain request-response cycle.